code/game/objects/items/flag.dm

```diff
@@ -267,8 +268,13 @@
 
 /obj/item/flag/chameleon/burn()
 	if(boobytrap)
-		boobytrap.prime()
-	..()
+		fire_act()
+		spawn(boobytrap.det_time)
+			boobytrap.loc = get_turf(loc)
+			boobytrap.prime()
+			..()
+	else
+		..()
```

Comment:
To avoid using spawns, I suggest something like this instead, using callbacks:
```dm
/obj/item/flag/chameleon/burn()
	if(boobytrap)
		fire_act()
		addtimer(CALLBACK(src, .proc/prime_boobytrap), boobytrap.det_time)
	return ..()
```
With the accompanying proc (name it whatever you want):
```dm
/obj/item/flag/chameleon/proc/prime_boobytrap()
	boobytrap.forceMove(get_turf(loc))
	boobytrap.prime()
```
Also `forceMove()` is preferred over setting `loc` directly.

Reply:
Echoing Steel here.
Some extra info
Spawns are less performent and also and are also not managed by a subsystem.
`forceMove` will also trigger quite some logic. Which ensures other objects and such react appropriately to the object in question.